src/goto-programs/show_goto_functions.cpp

```diff
@@ -52,32 +52,28 @@ void show_goto_functions(
   break;
 
   case ui_message_handlert::uit::PLAIN:
-    if(list_only)
     {
       for(const auto &fun : goto_functions.function_map)
       {
         const symbolt &symbol = ns.lookup(fun.first);
-        msg.status() << '\n'
-                     << symbol.display_name() << " /* " << symbol.name
-                     << (fun.second.body_available() ? ""
-                                                     : ", body not available")
-                     << " */";
-        msg.status() << messaget::eom;
-      }
-    }
-    else
-    {
-      auto &out = msg.status();
-      for(const auto &fun : goto_functions.function_map)
-      {
-        if(fun.second.body_available())
+
+        if(list_only)
+        {
+          msg.status() << '\n'
+                       << symbol.display_name() << " /* " << symbol.name
+                       << (fun.second.body_available() ? ""
+                                                       : ", body not available")
+                       << " */";
+          msg.status() << messaget::eom;
+        }
+        else if(fun.second.body_available())
         {
-          out << "^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n";
+          msg.status() << "^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n";
 
           const symbolt &symbol = ns.lookup(fun.first);
```

Comment:
Shouldn't this line be removed as it is now redundant?

Reply:
You mean because of the colouring?

Reply:
No, because of the refactoring in "consolidate the two variants of show-goto-functions".

Reply:
sorry, the line was unclear -- yes, redundant, yes, removed